### PR TITLE
Midi control refactoring

### PR DESCRIPTION
**Pull request by nittka**
In this series of commits the following issues are addressed:

* visually reset the position when the playback has finished
* move play/pause button from toolbar to the view area
* restructure and add more control buttons to the audio view: mark current position, reset to marked position
* make tempo slider hideable - changing the tempo will not be done on a regular basis
* improve the play/pause command; depending on the current view/selection the file to be played is changed, if the audio view is not yet open, it is opened

### CODE REVIEW

**Comment by nittka** [comment]
It would be nice, if you could have a look at this PR. I find being able to mark a position in the audio view very helpful for "proof-hearing". I'd love to have this feature in the next release.

File: org.eclipse.ui.views.midi/src/org/eclipse/ui/views/midi/MidiPlaybackControl.java [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+			public void widgetSelected(SelectionEvent e) {
+				rewind();
+				focusPlayButton();
+			}
+
+		});
+
+		Button marker = new Button(rowParent, SWT.FLAT);
+		marker.setImage(getImage("Mark"));
+		marker.setToolTipText("mark current position");
```
Comment (thSoft) [inline org.eclipse.ui.views.midi/src/org/eclipse/ui/views/midi/MidiPlaybackControl.java]:
```suggestion
		marker.setToolTipText("Mark current position");
```

**Review by thSoft — CHANGES_REQUESTED** [review]
Thanks for the enhancements, they are very useful! Please display the marker above/below the playback slider for better visual feedback.

File: org.eclipse.ui.views.midi/src/org/eclipse/ui/views/midi/MidiPlaybackControl.java (hunk 1 above)
Comment (nittka) [reply org.eclipse.ui.views.midi/src/org/eclipse/ui/views/midi/MidiPlaybackControl.java]: Done

**Comment by nittka** [comment]
I added a proof of concept commit for showing the marker position. The problem is that I have no idea how to obtain the sizes of the buttons (left, right, slider itself). Without these sizes - which may be operating system specific - the correct marker location cannot be calculated.

Two other problems I noticed:
* shift-ctrl-p has a key binding conflict with Xtext's go to matching bracket - do you have a suggestion for the play midi shortcut?
* the play action currently opens the view (and the correct file) only if the view's link-toggle is active; I'll have to check whether I can force opening the correct file independent of this toggle

**Comment by nittka** [comment]
I added a proposal for opening the audio view for a selected file that starts playing the midi even if the link-toggle is disabled. The  page!=null is necessary to prevent NPEs, obviously. Although showFile() is called (and a page is created), I had occasional null pointers, hence the three attempts.

(Once you are happy with the end result, I would clean up the commits before the actual merging)

**Review by thSoft — APPROVED** [review]
The marker position is not precise, but I think this is good enough. Thanks!

**Comment by nittka** [comment]
I did a last small cleanup.
* incorporating the waiting for opening a new file (toggle play if view is closed)
* disable the goto marker button if the marker is at the beginning of the file

If you don't object within a couple of days, I'll merge the PR.

**Pull request merged**